lib/puppet/parser/functions/fetch.rb

```diff
@@ -0,0 +1,70 @@
+module Puppet::Parser::Functions
+  newfunction(
+      :fetch,
```

Comment:
'fetch' is a very generic name. It's probably a good investment in the future to think hard about a more specific name.